ci/lint.sh

```diff
@@ -97,6 +97,19 @@ if [ "$LINT" ]; then
     fi
     echo "Check for use of lists instead of generators in built-in Python functions DONE"
 
+    echo "Check for use of comprehensions using set(), dict() or list()"
+
+    # Example: Avoid `list(i for i in some_iterator)` in favor of `[i for i in some_iterator]`
+    #
+    # Check the following functions:
```

Comment:
don''t you need to specify the directory?

e.g. on current master
```
(pandas) bash-3.2$ grep -R --include="*.py*" -E "(list|dict|[^frozen]set)\([^\{)=]* for .* in [^}]*\)" pandas
pandas/_libs/parsers.pyx:        dtypes = set(a.dtype for a in arrs)
pandas/_version.py:        tags = set(r for r in refs if re.search(r'\d', r))
pandas/core/dtypes/concat.py:        fill_values = set(c.fill_value for c in to_concat)
pandas/core/groupby.py:                            names = set(v.name for v in values)
pandas/core/indexes/base.py:        names = set(obj.name for obj in to_concat)
pandas/core/indexes/interval.py:        if not len(set(i.closed for i in to_concat if len(i))) == 1:
pandas/core/internals.py:        ndim = set(b.ndim for b in blocks)
pandas/core/internals.py:            if len(set(b.dtype for b in blocks)) != 1:
pandas/core/reshape/concat.py:            if not len(set(idx.nlevels for idx in indexes)) == 1:
pandas/io/parsers.py:        return set(i for i, name in enumerate(names)
pandas/io/pytables.py:        axis = list(set(t.non_index_axes[0][0] for t in tbls))[0]
pandas/tests/groupby/test_whitelist.py:    results = set(v for v in dir(grp) if not v.startswith('_'))
```

Reply:
I was wanting to lint the entire project (including python scripts in the other directories, e.g docs) and Travis was linting everything. 

I can change it to just lint the pandas directory if you want.

Reply:
no u need to specify a dir . is ok

the command doesn’t work atm

Reply:
Ah yes it looks my grep is too restrictive here.

Admittedly my grep-foo is a bit too weak for this lint. I am not sure how to write a grep to catching matching parenthesis, `list|set|dict(` to the matching `)`, and there are edge cases that are hard to catch, e.g. `list(set comprehension)` is okay, `dict()` can accept `=`  that is assigned to a list comprehension, etc. 

I can create a new issue for this unless someone with more grep experience can chime in.